docs/internal-architecture.md

```diff
@@ -0,0 +1,91 @@
+---
+description: Internal architecture of the project.
+---
+
+# Internal architecture
+
+### Trackers
+
+A tracker is usually in the type of an index tracker for a coin or token on a blockchain network.
+
+Index trackers take placed in the `pkg/tracker` package of the Telliot repository. the index tracker struct is as follow:
+
+```go
+type IndexTracker struct {
+	DB               db.DataServerProxy
+	Name             string
+	Identifier       string
+	Symbols          []string
+	Source           DataSource
+	Interval         time.Duration
+	Param            string
+	Type             IndexType
+	lastRunTimestamp time.Time
+}
+```
+
+an IndexTracker struct is related to an index object from the `indexes.json` file. based on `Source` there are two main types of trackers: HTTP trackers, on-chain trackers.
```

Comment:
I don't think we should add the actual internal implementation as this is another thing to maintain in the docs when the internal implementation changes. I think to have this docs as just a high level overview of the architecture.

Reply:
Got it, thanks.